base/applications/mspaint/dib.cpp

```diff
@@ -6,11 +6,14 @@
  * PROGRAMMERS: Benedikt Freisen
  */
 
-/* INCLUDES *********************************************************/
-
 #include "precomp.h"
 #include <math.h>
 
+INT fileSize = 0;
+INT fileHPPM = 2834; // horizontal pixels per millimeter (PPM)
+INT fileVPPM = 2834; // vertical pixels per millimeter (PPM)
```

Comment:
Where do these value come from?

Reply:
Win2k3 mspaint:
![image](https://user-images.githubusercontent.com/2107452/227682025-ab460c9d-256b-42a2-9b4e-fa9b6ed1ecf9.png)
101 dots per inch.

ReactOS mspaint on Win10:
![image](https://user-images.githubusercontent.com/2107452/227682053-264fa545-b0ab-4f40-985e-0eb46247cb09.png)
2834 pixels per meter. This value is wrong.

I have asked ChatGPT for it:

```txt
101 dots per inch = 101 dots / 1 inch
= 101 dots / 0.0254 meters (1 inch = 0.0254 meters)
= 3980.31 dots per meter
```

So we have to fix the `fileHPPM` and `fileVPPM` values.

```txt
1 dots per inch = 1 dots / 1 inch
= 1 dots / 0.0254 meters (1 inch = 0.0254 meters)
= 39.37 dots per meter
```

Reply:
`ATL` `CImage` class has no method to get the image resolution.
`gdiplus` `Image` class has `GetHorizontalResolution` and `GetVerticalResolution` methods.

Reply:
Now fixed.